pkg/osvscanner/osvscanner.go

```diff
@@ -87,6 +117,48 @@ func scanDir(r *output.Reporter, query *osv.BatchedQuery, dir string, skipGit bo
 	})
 }
 
+type gitIgnoreMatcher struct {
+	matcher  gitignore.Matcher
+	repoPath string
+}
+
+func parseGitIgnores(dir string) (*gitIgnoreMatcher, error) {
+	// We need to parse .gitignore files from the root of the git repo to correctly identify ignored files
+	// Defaults to current directory if dir is not in a repo or some other error
+	// TODO: Won't parse ignores if dir is not in a git repo, and is not under the current directory (e.g ../path/to)
```

Comment:
Can we create a bug to track this?   By "current directory", do you mean "current working directory"?

Reply:
Made #202